pkg/cloudclient/aws/aws.go

```diff
@@ -0,0 +1,212 @@
+package aws
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/sts"
+
+	"github.com/golang/glog"
+	awsv1alpha1 "github.com/openshift/aws-account-operator/pkg/apis/aws/v1alpha1"
+	awsclient "github.com/openshift/configuration-anomaly-detection/pkg/cloudclient/aws/client"
+	ccerrors "github.com/openshift/configuration-anomaly-detection/pkg/cloudclient/errors"
+	"github.com/openshift/configuration-anomaly-detection/pkg/ocm"
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
+	"github.com/pkg/errors"
+)
+
+const (
+	jumpRole      = "ba"
+	defaultRegion = "us-east-1" // TODO extract to own config
+	credentialDir = "~/.aws"    // TODO extract to own config
+)
+
+const (
+	// the ClusterDeployment label key we're expecting to be able to pull the ClusterID from
+	cdClusterIDLabelKey string = "api.openshift.com/id"
+)
+
+// CloudClient is the client used for handling AWS Cloud Handler requests
+type CloudClient struct {
+	// clusterDeployment is the ClusterDeployment for the target cluster (the one
+	// for which a login url is requested)
+	clusterDeployment *hivev1.ClusterDeployment
```

Comment:
Do we have to use the CD directly and not take what we exactly need?

Reply:
good point, I will try to define a ClusterDeployment Type in the ocm-client with only the values that are required.